api/specs.proto

```diff
@@ -270,4 +275,19 @@ message ClusterSpec {
 
 	// TaskDefaults specifies the default values to use for task creation.
 	TaskDefaults task_defaults = 7 [(gogoproto.nullable) = false];
+
+	// SecretConfig specifies cluster-level secret configuration settings, such as
+	// maximum number of secrets.
+	SecretsConfig secrets = 8 [(gogoproto.nullable) = false];
+}
+
+// SecretSpec specifies a user-provided secret.
+message SecretSpec {
+	Annotations annotations = 1 [(gogoproto.nullable) = false];
+
+	// Secret payload.
+	bytes data = 2;
+
+	// When this secret-spec should expire and no longer be available
+	Timestamp expiry = 3;
```

Comment:
If a user wanted an auto-expiring key that shouldn't be available after a certain period, for instance if the secret is the key to a cert that expires, they may want to ensure that services can no longer be started with said key.